thirdparty/faiss/faiss/IndexFlatElkan.h

```diff
@@ -29,9 +29,17 @@ namespace faiss {
 // Elkan algo was introduced into Knowhere in #2178, #2180 and #2258. 
 struct IndexFlatElkan : IndexFlat {
     bool use_elkan = true;
+    float* data = nullptr;
 
     explicit IndexFlatElkan(idx_t d, MetricType metric = METRIC_L2,
-                       bool is_cosine = false, bool use_elkan = true);
+                       bool is_cosine = false, bool use_elkan = true, float* data = nullptr);
```

Comment:
I see no point in passing `float* data` to constructor. By design, it is a temporary buffer

Reply:
Yes, remove it.